c10/util/StringUtil.h

```diff
@@ -87,6 +134,14 @@ inline decltype(auto) str(const Args&... args) {
   return detail::_str_wrapper<typename detail::CanonicalizeStrTypes<Args>::type...>::call(args...);
 }
 
+// Convert a list of error arguments into a single string by striping away string literal arguments, i.e.
+// if an argument is const char*, it will not show up in the returned message.
+// This is to optimize build size.
+template <typename... Args>
+inline decltype(auto) error_value(const Args&... args) {
```

Comment:
You probably want to make this unconditionally accept some number of leading arguments; otherwise you'll lose basic diagnostics like " INTERNAL ASSERT FAILED at"

Reply:
Had an offline discussion, because file, line information is already contained in error, we decided to only pass error arguments to message.